estate/models/estate_property.py

```diff
@@ -0,0 +1,100 @@
+from dateutil import relativedelta
+
+from odoo import _, api, exceptions, fields, models
+from odoo.tools import float_utils
+
+
+class EstateProperty(models.Model):
+    _name = 'estate.property'
+    _description = 'Estate property'
+    _order = 'id desc'
+
+    name = fields.Char(string='Name of the Property', required=True)
+    description = fields.Text(string='Description')
+    postcode = fields.Char(string='Postcode')
+    date_availability = fields.Date(
+        string='Available From',
+        copy=False,
+        default=fields.Date.today() + relativedelta.relativedelta(months=3),
+    )
+    expected_price = fields.Float(string='Expected Price', required=True)
+    selling_price = fields.Float(string='Selling Price', readonly=True, copy=False)
+    bedrooms = fields.Integer(string='Number Bedrooms', default=2)
+    living_area = fields.Integer(string='Living Area (sqm)')
+    facades = fields.Integer(string='Facades')
+    garage = fields.Boolean(string='Garage')
+    garden = fields.Boolean(string='Garden')
+    garden_area = fields.Integer(string='Garden Area (sqm)')
+    garden_orientation = fields.Selection(
+        string='Garden Orientation',
+        selection=[('north', 'North'), ('south', 'South'), ('east', 'East'), ('west', 'West')],
+    )
+    active = fields.Boolean(default=True)
+    state = fields.Selection(
+        string='State',
+        selection=[
+            ('new', 'New'),
+            ('received', 'Offer Received'),
+            ('accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('cancelled', 'Cancelled'),
+        ],
+        required=True,
+        copy=False,
+        default='new',
+    )
+    property_type_id = fields.Many2one('estate.property.type', string='Property Type')
+    users_id = fields.Many2one('res.users', string='Salesman', default=lambda self: self.env.user)
+    partner_id = fields.Many2one('res.partner', string='Buyer', readonly=True)
+    tag_ids = fields.Many2many('estate.property.tag', string='Property Tags')
+    offer_ids = fields.One2many('estate.property.offer', 'property_id', string='Offers')
+    total_area = fields.Integer(string='Total Area (sqm)', compute='_compute_total_area')
+    best_price = fields.Integer(string='Best Offer', compute='_compute_price')
+
+    _sql_constraints = [
+        ('check_expected_price', 'CHECK(expected_price > 0)', 'The expected price must strictly be positive.'),
+        ('check_selling_price', 'CHECK(selling_price > 0)', 'The selling price must be positive.'),
+    ]
+
+    @api.ondelete(at_uninstall=False)
+    def prevent_delete_based_on_state(self):
+        if any(record.state in {'new', 'cancelled'} for record in self):
+            raise exceptions.UserError(_('A new or cancelled property cannot be deleted.'))
+
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+        for record in self:
+            record.write({'total_area': record.living_area + record.garden_area})
```

Comment:
I find write less easy too read and less pythonic
```suggestion
            record.total_area = record.living_area + record.garden_area
```